components/raftstore/src/store/fsm/peer.rs

```diff
@@ -637,6 +638,7 @@ where
                         match self.fsm.peer.maybe_destroy(self.ctx) {
                             None => self.ctx.raft_metrics.message_dropped.applying_snap += 1,
                             Some(job) => {
+                                // [PerformanceCriticalPath]?? could it be done async?
```

Comment:
How about leaving a TODO instead since it's stated at the header of the file. And it should be added at the definition of `destroy_peer` instead of spreading everywhere.

Reply:
sounds good.